be/src/exec/vectorized/json_scanner.cpp

```diff
@@ -459,52 +457,121 @@ Status JsonReader::_read_rows(Chunk* chunk, int32_t rows_to_read, int32_t* rows_
     return Status::OK();
 }
 
-Status JsonReader::_construct_row(simdjson::ondemand::object* row, Chunk* chunk,
-                                  const std::vector<SlotDescriptor*>& slot_descs) {
-    if (_scanner->_json_paths.empty()) {
-        // No json path.
-
-        for (SlotDescriptor* slot_desc : slot_descs) {
-            if (slot_desc == nullptr) {
+Status JsonReader::_construct_row_in_object_order(simdjson::ondemand::object* row, Chunk* chunk) {
```

Comment:
How man performance degradation when the columns' number dismatch?

Reply:
> ## Benchmark
> 
> load type: Stream Load source file: 65MB in total, json array format, 6KB / object, 64 fields /object table: 330 columns, 10 buckets
> version 	throughput
> main 	2.5MB/s
> [c8d00b2](https://github.com/StarRocks/starrocks/commit/c8d00b2e40324b21a0143cd0813fca71aae20b6f)@perf/sparse-column 	18.2MB/s

I think this could be the answer. The json input has 64 fields and the target table has 300 columns.